examples/react/large-file-based/src/routeTree.gen.ts

```diff
@@ -127,18 +127,97 @@ declare module '@tanstack/react-router' {
+}
+const ParamsRouteRouteWithChildren = ParamsRouteRoute.addChildren(
+  ParamsRouteRouteChildren,
+)
+const SearchSearchPlaceholderRouteWithChildren = SearchSearchPlaceholderRoute
+interface SearchRouteRouteChildren {
+  SearchSearchPlaceholderRoute: typeof SearchSearchPlaceholderRouteWithChildren
+}
+const SearchRouteRouteChildren: SearchRouteRouteChildren = {
+  SearchSearchPlaceholderRoute: SearchSearchPlaceholderRouteWithChildren,
+}
+const SearchRouteRouteWithChildren = SearchRouteRoute.addChildren(
+  SearchRouteRouteChildren,
+)
+const AbsoluteRouteWithChildren = AbsoluteRoute
+const LinkPropsRouteWithChildren = LinkPropsRoute
+const RelativeRouteWithChildren = RelativeRoute
+
+interface FileRoutesByFullPath {
+  '/': typeof IndexRouteWithChildren
+  '/params': typeof ParamsRouteRouteWithChildren
+  '/search': typeof SearchRouteRouteWithChildren
+  '/absolute': typeof AbsoluteRouteWithChildren
+  '/linkProps': typeof LinkPropsRouteWithChildren
+  '/relative': typeof RelativeRouteWithChildren
+  '/params/$paramsPlaceholder': typeof ParamsParamsPlaceholderRouteWithChildren
+  '/search/searchPlaceholder': typeof SearchSearchPlaceholderRouteWithChildren
+}
+
+interface FileRoutesByTo {
+  '/': typeof IndexRouteWithChildren
+  '/params': typeof ParamsRouteRouteWithChildren
+  '/search': typeof SearchRouteRouteWithChildren
+  '/absolute': typeof AbsoluteRouteWithChildren
+  '/linkProps': typeof LinkPropsRouteWithChildren
+  '/relative': typeof RelativeRouteWithChildren
+  '/params/$paramsPlaceholder': typeof ParamsParamsPlaceholderRouteWithChildren
+  '/search/searchPlaceholder': typeof SearchSearchPlaceholderRouteWithChildren
+}
+
+interface FileRouteTypes {
+  fileRoutesByFullPath: FileRoutesByFullPath
+  fullPaths:
+    | '/'
+    | '/params'
+    | '/search'
+    | '/absolute'
+    | '/linkProps'
+    | '/relative'
+    | '/params/$paramsPlaceholder'
+    | '/search/searchPlaceholder'
+  fileRoutesByTo: FileRoutesByTo
+  to:
+    | '/'
+    | '/params'
+    | '/search'
+    | '/absolute'
+    | '/linkProps'
+    | '/relative'
+    | '/params/$paramsPlaceholder'
+    | '/search/searchPlaceholder'
+}
+
+interface RootRouteChildren {
+  IndexRoute: typeof IndexRouteWithChildren
+  ParamsRouteRoute: typeof ParamsRouteRouteWithChildren
+  SearchRouteRoute: typeof SearchRouteRouteWithChildren
+  AbsoluteRoute: typeof AbsoluteRouteWithChildren
+  LinkPropsRoute: typeof LinkPropsRouteWithChildren
+  RelativeRoute: typeof RelativeRouteWithChildren
+}
+
+const rootRouteChildren: RootRouteChildren = {
+  IndexRoute: IndexRouteWithChildren,
+  ParamsRouteRoute: ParamsRouteRouteWithChildren,
+  SearchRouteRoute: SearchRouteRouteWithChildren,
+  AbsoluteRoute: AbsoluteRouteWithChildren,
+  LinkPropsRoute: LinkPropsRouteWithChildren,
+  RelativeRoute: RelativeRouteWithChildren,
+}
+
+export const routeTree = rootRoute
+  .addChildren(rootRouteChildren)
+  ._addFileTypes<FileRouteTypes>()
```

Comment:
I've taken this approach so I can write type tests for it. Effectively I'm attaching additional types to the route tree which will short circuit types for later

Reply:
any downsides of adding those types just for the tests? e.g. does it have performance implications ? 
we could only add them conditionally when building / running the tests

Reply:
Nah, no performance implications from what I can tell